lib/_stream_readable.js

```diff
@@ -647,7 +660,8 @@ function maybeReadMore_(stream, state) {
   //   called push() with new data. In this case we skip performing more
   //   read()s. The execution ends in this method again after the _read() ends
   //   up calling push() with more data.
-  while (!state.reading && !state.ended &&
+  while (!state.reading && !state.ended && !state.destroyed &&
+         state.constructed &&
```

Comment:
Can we move the check for `state.constructed` in the `maybeReadMore`  method? There is no reason to check this in a while-loop, as this cannot change while reading.

Reply:
Good idea. Fixed.